travel-agency-service/src/main/java/cz/muni/fi/pa165/bluebat/service/CustomerServiceImpl.java

```diff
@@ -26,16 +27,17 @@ public CustomerServiceImpl(CustomerDao customerDao) {
 
     @Override
     public void deleteCustomer(Customer customer) {
-        if (customer == null) {
-            throw new IllegalArgumentException("Customer can not be null");
-        }
-
-        Customer previous = customerDao.findById(customer.getId());
+        Validator.NotNull(customer,"Customer");
+        Customer previous;
 
-        if (previous == null) {
-            throw new IllegalArgumentException("Customer has not been found in database");
+        try {
+            previous = customerDao.findById(customer.getId());
+        } catch (Exception e) {
+            throw new WrongDataAccessException("Customer dao layer exception", e);
         }
 
+        Validator.Found(previous, "Customer");
+
```

Comment:
Zbytečně při volání mazání uživatele přistupuješ dvakrát do DB (poprvé ve facade a podruhé tady na řádku 34). Celý tento blok mi přijde zbytečný. Pokud je totiž null, tak to projde a nic se nestane, pokud už v DB není, tak se taky nic nestane.